### DISCUSSION TITLE
Adding Raylib frontend to Lem

### DISCUSSION THREAD

**Discussion by ghost** [lem-project/lem#835] [General]
This is just an idea but how about we add a simpler frontend like Raylib[for now]? This would allow us to easily write graphics code at a higher level than SDL2. The core goal here is to make writing our custom widgets and scenes in lem easier. 
Raylib offers a far easier and a  higher level  api when compared to SDL2 and also offers very good compute shader support,which will allow us to develop applications and games inside the IDE, making this a far more power IDE than any other. 

Raylib source code is also far more readable and the API is cleanly divided, which means rewriting components of it in common lisp for some or the other reasons will not be difficult, this is not possible with sdl2 because it is hellish to read and understand. 

The end goal, as per me, should be removal of as much C code as possible to enable and provide the users with utmost level of hackability.

**Comment by Sasanidas** [comment]
I don't have any experience with Raylib, but feel free to test it out with Lem and see how it works, for now there are not that many documentation, but you can always check the code on the /frontends directory and here for some display concepts https://lem-project.github.io/lem-page/development/displaying/
